### ISSUE TITLE
Timeout when waiting for 3rd party check iframe message.

### ISSUE THREAD

**Issue by alidandach** [keycloak/keycloak#14414]
### Describe the bug

Just I have a spine "loading the admin console" when I try to access the login form **http://myip:8080/admin/master/console/**, 

### Version

19.0.1

### Expected behavior

The login form should appear

### Actual behavior

Just a spine with a message says "loading the admin console" 

### Anything else?

![Screenshot from 2022-09-14 20-05-26](https://user-images.githubusercontent.com/35876342/190218050-44d0d1ee-a408-44a1-bbdc-39c2e6b81f2f.png)

**Comment by lexcao** [comment]
Could you please provide more info to help me debug?
* What's your start command
* What's your Keycloak configuration

**Comment by alidandach** [comment]
I am using systemd to run this service, so I have attached the service file with the configuration in the same zip file
[keycloak-systemd-configuration.zip](https://github.com/keycloak/keycloak/files/9569252/keycloak-systemd-configuration.zip)

**Comment by BlueRoyal** [comment]
I am running into the same issue while using docker.

**Comment by pedroigor** [comment]
@alidandach @BlueRoyal I tried running Keycloak using systemd. I was able to access the administration console.

When running in dev mode, it should be fine to call the server using `localhost`, `myip` (if you are setting via `/etc/hosts`), or using the node IP.

**Comment by stianst** [comment]
Could you include results from the network tab in the developer console?

**Comment by lexcao** [comment]
Hi, @LeVraiRoiDHyrule 
Thanks for your screenshot. 
I can see there are many 200 CSS requests but no clue what the error is.

Could you please include a screenshot of the error requests?

**Comment by pedroigor** [comment]
Closing this issue as it should be related to https://github.com/keycloak/keycloak/issues/14466. If not, feel free to re-open.

**Comment by Shishir53** [comment]
Hi,

I have faced this issue with 19.0.3, not sure if it's fixed in this release.

Can you confirm @pedroigor ?

**Comment by danielclough** [comment]
# Solved

I am having this issue unrelated to Firefox, but related to 3rd party cookies.
Perhaps my debugging efforts may be helpful to someone.

## TLDR;
> I am running keycloak in docker using Traefik as a reverse proxy.
> I got it working by starting with this command:
```
start-dev --hostname=keycloak.domain.tld --proxy-headers=xforwarded --hostname-strict=false
```

## Debugging steps:

Using Chromium I see these errors :
```
Request URL:
http://keycloak.domain.tld/realms/master/protocol/openid-connect/3p-cookies/step1.html
Referrer Policy:
strict-origin-when-cross-origin
```

```
Refused to frame 'http://keycloak.domain.tld/' because it violates the following Content Security Policy directive: "frame-src 'self'".

Mixed Content: The page at 'https://keycloak.domain.tld/admin/master/console/' was loaded over HTTPS, but requested an insecure resource 'http://keycloak.domain.tld/realms/master/protocol/openid-connect/3p-cookies/step1.html'. This request has been blocked; the content must be served over HTTPS.

console/:1 Uncaught 
{error: 'Timeout when waiting for 3rd party check iframe message.'}
...
```

So, that looks like [`getRealmUrl()`](https://github.com/keycloak/keycloak/blob/be32f8b1bfb4fa709aa48552a1bbfdcf9fab6299/js/libs/keycloak-js/src/keycloak.js#L849) is not upgrading my connection to https.
 
---

Trying to configure hostname using `[bash /opt/keycloak/bin/kc.sh start --hostname=keycloak.domain.tld](https://www.keycloak.org/server/hostname)
returned error:
```
ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) ERROR: Key material not provided to setup HTTPS. Please configure your keys/certificates or start the server in development mode.
```

---

I am using Traefik to configure my SSL.

I notice [docs for setting up a proxy](https://www.keycloak.org/server/hostname) and start with the command:
```
start-dev --hostname=keycloak.domain.tld --proxy-headers=xforwarded --hostname-strict=false
```

And, it works!

**Comment by Leandro4002** [comment]
With newer version of keycloak, the "proxy-headers" parameter has been modified. Instead of
`start-dev --hostname=keycloak.domain.tld --proxy-headers=xforwarded --hostname-strict=false`

it is:
`start-dev --hostname=keycloak.domain.tld --proxy=passthrough --hostname-strict=false`

**Comment by stealthemoon0331** [comment]
I have set but still cant loading page.

![Image](https://github.com/user-attachments/assets/c6003be2-8e1a-4f4a-ae4d-4820ecc0184d)

![Image](https://github.com/user-attachments/assets/bc804f11-cee5-4623-b6df-7a38baa4b1c8)

**Comment by jedahan** [comment]
Same issue. If I set `KC_HOSTNAME_STRICT: "false"`, I get a warning saying it is ignored when setting a hostname, and then these errors during startup:

```
42,066 ERROR [org.jgroups.protocols.TCP] (pd-bundler-5,eeb914bcc7ad-19963) eeb914bcc7ad-19963: failed sending message to 948d2087b063-28783: java.net.ConnectException: Connection refused
[keycloak]          | 2025-07-13 15:05:42,251 WARN  [org.jgroups.protocols.pbcast.GMS] (main) eeb914bcc7ad-19963: JOIN(eeb914bcc7ad-19963) sent to 948d2087b063-28783 timed out (after 2000 ms), on try 6
[keycloak]          | 2025-07-13 15:05:42,253 ERROR [org.jgroups.protocols.TCP] (pd-bundler-5,eeb914bcc7ad-19963) eeb914bcc7ad-19963: failed sending message to 948d2087b063-28783: java.net.ConnectException: Connection refused
[keycloak]          | 2025-07-13 15:05:42,467 ERROR [org.jgroups.protocols.TCP] (pd-bundler-5,eeb914bcc7ad-19963) eeb914bcc7ad-19963: failed sending message to 948d2087b063-28783: java.net.ConnectException: Connection refused
[keycloak]          | 2025-07-13 15:05:42,667 ERROR [org.jgroups.protocols.TCP] (pd-bundler-5,eeb914bcc7ad-19963) eeb914bcc7ad-19963: failed sending message to 948d2087b063-28783: java.net.ConnectException: Connection refused
[keycloak]          | 2025-07-13 15:05:42,867 ERROR [org.jgroups.protocols.TCP] (pd-bundler-5,eeb914bcc7ad-19963) eeb914bcc7ad-19963: failed sending message to 948d2087b063-28783: java.net.ConnectException: Connection refused
[keycloak]          | 2025-07-13 15:05:43,068 ERROR [org.jgroups.protocols.TCP] (pd-bundler-5,eeb914bcc7ad-19963) eeb914bcc7ad-19963: failed sending message to 948d2087b063-28783: java.net.ConnectException: Connection refused
[keycloak]          | 2025-07-13 15:05:43,268 ERROR [org.jgroups.protocols.TCP] (pd-bundler-5,eeb914bc
```
